mmdet/core/fp16/utils.py

```diff
@@ -0,0 +1,159 @@
+from collections import abc
+from inspect import getfullargspec
+
+import numpy as np
+import torch
+
+
+# copy updated param from fp32_weight to fp16 net
+def copy_in_params(fp16_net, fp32_weight):
+    for net_param, fp32_weight_param in zip(fp16_net.parameters(),
+                                            fp32_weight):
+        net_param.data.copy_(fp32_weight_param.data)
+
+
+# copy gradient from fp16 net to fp32 weight copy
+def set_grad(fp16_net, fp32_weight):
+    for param, param_w_grad in zip(fp32_weight, fp16_net.parameters()):
+        if param_w_grad.grad is not None:
+            if param.grad is None:
+                param.grad = param.data.new(*param.data.size())
+            param.grad.data.copy_(param_w_grad.grad.data)
+
+
+def convert(inputs, src_type, dst_type, min_dim=0):
```

Comment:
Do not use ambiguous names such as `convert`. A better alternative is `cast_tensor_type`.